cadcutils/cadcutils/util/utils.py

```
@@ -172,43 +173,155 @@ def parse_resource_id(resource_id):
+    up in each subcommand (and not on the parent parser). Without subparsers (subparsers=False)
+    the common options are automatically added to the base parser.
+
+    :param subparsers: True if the parser will use subparsers (subcommands) otherwise False
     :param version: A version number if desired.
-    :param usecert: If True add '--certfile' argument.
+    :param usecert: If True add '--cert' argument.
     :param default_resource_id: default resource identifier to use
     :return: An ArgumentParser instance.
     """
-    parser = ArgumentParser(add_help=False)
+    cparser = ArgumentParser(add_help=False, formatter_class=SingleMetavarHelpFormatter)
+
+    if version is not None:
+        cparser.add_argument('-V', '--version', action='version', version=version)
+    auth_group = cparser.add_mutually_exclusive_group()
     if usecert:
-        parser.add_argument('--certfile', type=str,
-                            help="location of your CADC certificate "
-                            + "file (default: $HOME/.ssl/cadcproxy.pem, " +
-                            "otherwise uses $HOME/.netrc for name/password)",
-                            default=os.path.join(os.getenv("HOME", "."),
-                                                 ".ssl/cadcproxy.pem"))
-    parser.add_argument('--anonymous', action="store_true",
-                        help='Force anonymous connection')
-    parser.add_argument('--host', help="Base hostname for services - used mainly for testing " +
-                                       "(default: www.cadc-ccda.hia-iha.nrc-cnrc.gc.ca)",
+        auth_group.add_argument('--cert', type=str,
+                            help='location of your X509 certificate to use for authentication ' +
+                            '(unencrypted, in PEM format)')
+    auth_group.add_argument('-n', action='store_true', help='Use .netrc in $HOME for authentication')
```

If you talk to Pat, he's going to tell you that he hates applications do magic (silently using network and password resources). And the curl guys seem to agree with him since this is also the CLI they use. That's not to say that I sympathize with your point of view.
A good compromise IMO, is to have these authentication arguments (username, cert, netrc) explicit but give users (power users) the option to shortcut them by adding these options to an application config file. This way the application doesn't do any magic but the users, understanding the requirements, can customize the arguments themselves.
Should we take it to a vote?